qutebrowser/mainwindow/tabwidget.py

```diff
@@ -158,6 +167,7 @@ def addTab(self, page, icon_or_text, text_or_empty=None):
             text = text_or_empty
             new_idx = super().addTab(page, icon, '')
         self.set_page_title(new_idx, text)
+        page.scroll_pos_changed.connect(lambda: self.update_tab_title(new_idx))
```

Comment:
This should be in `qutebrowser/mainwindow/tabbedbrowser.py` in `_connect_tab_signals` instead. Then you also only have to do it in one place.